### DISCUSSION TITLE
Setting an extra cookie and using it to control the consent form

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#4674] [Q&A] [answered]
I'm trying to get cookie consent to be shared across docs and our main website. Our main website sets a cookie, `n8n-consent`, with the value `{'consent': true}` when a user agrees to cookies.

I've managed to get it to work from docs -> website. In other words, when a user accepts cookies on the docs website, it also sets the `n8n-consent` cookie, and our main website picks that up.

I've _almost_ got it working the other way, but not quite. I get this behaviour: 

accept cookies on main website -> go to docs website -> consent banner displays -> refresh docs page (don't click anything, just refresh) -> consent banner now doesn't display

My best guess at the problem is that the consent banner code runs & displays before my extra JS has had a chance to read the `n8n-consent` cookie and set `__consent` (the `md` setting in local storage that remembers cookie settings). One workaround seemed to be to hide the banner again in my JS. However, because the Material consent code includes a timeout, it overrides my attempt to hide the consent banner in extra JS (and I am concerned that even if I got this working there could be a flicker if the consent form briefly displayed)

Relevant Material code: https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/javascripts/consent.html

My code is here: https://github.com/n8n-io/n8n-docs/blob/main/docs/_extra/javascript/extra.js

Before I started writing my own JS for this, I did try simply renaming the cookie, as described [here](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#cookie-consent). But this didn't seem to do anything. Material's consent seems to be stored in local storage, rather than in a cookie.

So: anyone know a way to get extra JS to run before Material's own? Or some other approach to ensure the consent banner is hidden? Any advice appreciated!

**Comment by squidfunk** [comment, marked as answer]
Yes, Material for MkDocs' cookie consent form is not really setting cookies but uses local storage. Your use case is a little exotic, which is why [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) might be your best option. You could override the [`javascripts/consent`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/javascripts/consent.html) partial and apply your logic _before_ the theme's logic runs. Extra JavaScript will always run after the theme's logic. The consent logic is here (bottom):

https://github.com/squidfunk/mkdocs-material/blob/91bc63cfd87b9e08e4fddaf92191db12a228a144/src/base.html#L359-L371

**Reply by StarfallProjects** [reply]
Thank you! Working 😀

> Your use case is a little exotic

You're very polite 😛

**Reply by ssbarnea** [reply]
Indeed, very polity. The moment I seent the "setting cookie" I got scared. Please don't bring the cookie into the discussions, everyone should use "local storage" and stop sending data to web servers.
